axelrod/ecosystem.py

```diff
@@ -0,0 +1,38 @@
+class Ecosystem(object):
+    """Create an ecosystem based on the payoff matrix from an Axelrod tournament."""
+
+    def __init__(self, pmatrix):
+
+        self.pmatrix = pmatrix
+        self.nplayers = len(self.pmatrix)
+
+        # Population histories will be recorded in a nested list,
+        # with each internal list containing a list of populations
+        # for all strategies. The first list, representing the
+        # starting populations, should have all equal values.
+        self.populations = [[1.0 / self.nplayers for i in range(self.nplayers)]]
+
+    def reproduce(self, turns):
+
+        for iturn in range(turns):
+
+            plist = list(range(self.nplayers))
+            pops = self.populations[-1]
+
+            # The unit payoff for each player in this turn is a sum of the payoffs
+            # obtained from all other players scaled by the size of the opponent population.
+            payoffs = [sum([self.pmatrix[ip][jp] * pops[jp] for jp in plist]) for ip in plist]
+
+            # In our case lower payoffs are better, so fitness should increase when
+            # the payoff descreases. This is a simple linear function that fulfills
+            # this and still behaves nicely, but we could chooise more complicated forms.
+            pmax = 5 * self.nplayers
```

Comment:
It could be, but it is the function used to get fitness which is the main unknown. So it would make most sense to make that a variable. I don't know what the "best" choice there is, but I think it won't really matter and will affect only kinetics. As long as the function is monotonic. I could be wrong!